viewer/core/src/public/migration/restrictViewerOptionsToDevice.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Cognite3DViewerOptions } from './types';
+
+import log from '@reveal/logger';
+import { assertNever, isMobileOrTablet } from '@reveal/utilities';
+
+/**
+ * Modifies options to disable features we don't support on the current device. The following options might be modified:
+ * - SSAO - disabled for mobile/tablet
+ * - Anti-aliasing - for mobile/tablet we will fall back to FXAA when MSAA is enabled.
+ * @param options
+ */
+export function restrictViewerOptionsToDevice(options: Cognite3DViewerOptions): void {
+  options.antiAliasingHint = restrictAntialiasOption(options.antiAliasingHint);
+  options.ssaoQualityHint = restrictSsaoOption(options.ssaoQualityHint);
+}
+
+type AntiAliasingHint = Cognite3DViewerOptions['antiAliasingHint'] | undefined;
+type SsaoQualityHint = Cognite3DViewerOptions['ssaoQualityHint'] | undefined;
+
+function restrictAntialiasOption(antiAliasingHint: AntiAliasingHint): AntiAliasingHint {
+  if (!isMobileOrTablet()) {
+    return antiAliasingHint;
+  }
+
+  switch (antiAliasingHint) {
+    case 'msaa2':
+    case 'msaa4':
+    case 'msaa8':
+    case 'msaa16':
+    case 'msaa2+fxaa':
+    case 'msaa4+fxaa':
+    case 'msaa8+fxaa':
+    case 'msaa16+fxaa':
+      log.warn(`Anti-aliasing mode '${antiAliasingHint}' is not supported on mobile devices, falling back to 'fxaa'`);
+      return 'fxaa';
```

Comment:
msaaXX (without fxaa) will fall through and return 'fxaa'. I suppose this is unintended?

Reply:
This was intended - but we can change if you'd like. My logic was that if user wanted msaa and we can't enable it, we should at least enable fxaa.